packages/test/local-server-stress-tests/src/localServerStressHarness.ts

```diff
@@ -1338,3 +1351,111 @@ export namespace createLocalServerStressSuite {
 			});
 }
+
+/**
+ * Operation type for cloning a client using its pending local state.
+ *
+ * This operation triggers a clone of an existing client by capturing its
+ * pending local state (via `getPendingLocalState()`) and loading a new
+ * container from it.
+ */
+interface GetClientPendingState {
+	type: "getClientPendingState";
+	sourceClientTag: `client-${number}`;
+	newClientTag: `client-${number}`;
+}
+
+/**
+ * Adds support for "getPendingLocalState"-based client cloning to the stress model.
+ *
+ * This mixin extends the base fuzz model by occasionally inserting operations that:
+ * - Capture a given client's pending local state.
+ * - Instantiate a new client from that state using the same container URL.
+ * - Add the cloned client to the test state for further fuzzing.
+ *
+ * Preconditions:
+ * - Must have at least one client already in the system.
+ * - Must be in an attached container state.
+ */
+function mixinGetClientPending<TOperation extends BaseOperation>(
+	model: LocalServerStressModel<TOperation>,
+	options: LocalServerStressOptions,
+): LocalServerStressModel<TOperation | GetClientPendingState> {
+	const generatorFactory: () => AsyncGenerator<
+		TOperation | GetClientPendingState,
+		LocalServerStressState
+	> = () => {
+		const baseGenerator = model.generatorFactory();
+		return async (
+			state: LocalServerStressState,
+		): Promise<TOperation | GetClientPendingState | typeof done> => {
+			const { clients, random, validationClient } = state;
+			assert(
+				state.validationClient.container.attachState === AttachState.Attached,
+				"Validation client must be attached before getting pending state",
+			);
+			if (
+				options.clientJoinOptions !== undefined &&
+				validationClient.container.attachState !== AttachState.Detached &&
+				clients.length > 0
+			) {
+				return {
+					type: "getClientPendingState",
+					sourceClientTag: random.pick(clients).tag,
+					newClientTag: state.tag("client"),
+				} satisfies GetClientPendingState;
+			}
+			return baseGenerator(state);
+		};
+	};
+
+	const minimizationTransforms: MinimizationTransform<TOperation | GetClientPendingState>[] =
+		(model.minimizationTransforms as
+			| MinimizationTransform<TOperation | GetClientPendingState>[]
+			| undefined) ?? [];
+
+	const reducer: AsyncReducer<
+		TOperation | GetClientPendingState,
+		LocalServerStressState
+	> = async (state, op) => {
+		if (isOperationType<GetClientPendingState>("getClientPendingState", op)) {
+			const sourceClientIndex = state.clients.findIndex((c) => c.tag === op.sourceClientTag);
+			assert(sourceClientIndex !== -1, `Client ${op.sourceClientTag} not found`);
+			const sourceClient = state.clients[sourceClientIndex];
+
+			assert(
+				typeof sourceClient.container.getPendingLocalState === "function",
+				`Client ${op.sourceClientTag} does not support getPendingLocalState`,
+			);
+
+			const pendingLocalState = await sourceClient.container.getPendingLocalState?.();
```

Comment:
With the assert above, this shouldn't need the null check
```suggestion
			const pendingLocalState = await sourceClient.container.getPendingLocalState();
```